cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -175,9 +187,34 @@ func (p *Planner) injectOngoingActuation() error {
 	if err != nil {
 		return err
 	}
-	// TODO(x13n): Check owner references to avoid double-counting already
-	// recreated pods.
-	return p.injectPods(p.actuationStatus.RecentEvictions())
+	return p.injectPods(filterOutRecreatedPods(p.actuationStatus.RecentEvictions(), p.cc))
+}
+
+func filterOutRecreatedPods(pods []*apiv1.Pod, cc controllerReplicasCalculator) []*apiv1.Pod {
+	var podsToInject []*apiv1.Pod
+	addedReplicas := make(map[string]int32)
+	for _, pod := range pods {
+		ownerRef := getSupportedOwnerRef(pod.GetOwnerReferences())
+		if ownerRef == nil {
+			podsToInject = append(podsToInject, pod)
```

Comment:
If it's not one of the supported owner kinds, it probably won't get recreated right (e.g. just a naked pod)? Are we assuming it could get recreated by some custom controller to be on the safe side? If so, could you clarify that in a comment?

Reply:
And should we distinguish static/mirror pods? They won't have a supported owner ref, but they won't get recreated (just like DS pods).

Reply:
addressed that using filterRecreatable

Reply:
I have a couple of comments but all tied to this chain so I'll gather them here:

1. IMO it'd still be very much non-obvious to a future reader why we're choosing to inject the pods (and not disregard) in case of uncertainty/errors. A comment clarifying that injecting the pod actually means being on the safe side (and why) would help a lot. Some of it is mentioned in the method comment for `injectOngoingActuation` (which got outdated with this change btw), so maybe we could just update that, or move that comment here.

2. Hmm, but isn't `filterRecreatable` called only for `currentlyDrainedPods` and not `RecentEvictions`? So right now, won't we inject DS/static/mirror pods returned by `RecentEvictions` when we should't?

3. Also, I'd remove DaemonSet as known ref in `getKnownOwnerRef` - then ownerRef will be nil for all DS/static/mirror pods, they will be added here, but then filtered out by `filterRecreatable` down the line. Because right now I think DS pods will have a non-nil ref, but `getReplicas` will return an error - which seems like doing a lot of extra steps, and differs from how static/mirror pods are handled.

4. Do we have a test case for filtering DS/static/mirror pods? Seems like something should've started failing after you forgot the `filterRecreatable` call, but the CI looks like everything passed. Maybe it'd be worth adding a test case for that?

Reply:
1. Added a comment here in `filterOutRecreatedPods` and updated comment in `injectOngoingActuation`
2. It was actually called for pods in `filterOutRecreatedPods`  at the beginning of function, though I moved it to `injectOngoingActuation` to make it consistent with `currentlyDrainedPods`
3. Removed DaemonSet from `getKnownOwnerRef`
4. I added test case with DS, though as stated in point 2, the `filterRecreatable` was run in previous revision as well

Reply:
Thanks, I knew I must've missed something :P Everything looks clear now, thanks!